DOCS-10044 Anypoint Security Jan 20 Release (#54)
mulesoft/docs-anypoint-security: modules/ROOT/pages/asm-secret-group-creation-task.adoc

= Create and Edit Secret Group (Anypoint Platform)

> Using Secrets Manager you can create a secrets group and add secret types to it.

Reviewer: Using Secrets Manager you can create a secrets group, a logical grouping of secrets, and add secret types to it.

Author: Rewrote as:

Secrets Manager enables you to create and edit secrets groups, which is a logical grouping of secrets. Secrets Manager also enables you to create new secret types and add them to a secrets group.

= Create and Edit Secret Group (Anypoint Platform)

> Secrets Manager enables you to create and edit secrets groups, which is a logical grouping of secrets. Secrets Manager also enables you to create new secret types and add them to a secrets group.

Reviewer 2: Out of scope but something to think about: could create a partial here so you're not having to restate what a secret group is.

== Prerequisites

To create and edit secret groups:

* Verify that you have the `Write secrets` permission enabled. 
* Verify that you are in the correct environment using the environment or business group.

== Create a Secret Group

. In Anypoint Platform, go to Management Center then select *Secrets Manager*.
. Click *Create Secret Group*.
. Enter a name for the secret group and click *Save*.
+
The name of your secret group must: 
+
* Start with a letter
* Be at least three characters long and no longer than 35 characters

> * Contain only letters, numbers, and dashes, however, the name cannot end with a dash.

Reviewer 2: ```suggestion
* Contain only letters, numbers, and dashes; however, the name cannot end with a dash.
```

> After creating your secret group, it appears in the *Secret Groups* list view. Edit the secret group to add the necessary secret types, for example adding a truststore or a keystore.

Reviewer 2: ```suggestion
After creating your secret group, it appears in the *Secret Groups* list view. Edit the secret group to add the necessary secret types, such as a truststore or a keystore.
```

Reviewer 2: Take this comment with a grain of salt; "adding" seemed redundant, but I don't have the context to know that for sure.

== Edit a Secret Group

> Edit a secret group to add secret types such as Keystore, Truststore, and Certificates.

Reviewer 3: can we change from "to add" to "to manage", also after Certificates, please add " etc"?
Edit a secret group to manage secret types such as Keystore, Truststore, Certificates and etc.

Author: Changed.

. In the *Secret Groups* list view, select the secret group you want to edit, and click *Edit*. 

> . Make your changes as required, then click *Finish.*

Reviewer 4: Can we remove this line? 'Make your changes as required, then click *Finish.*'. The finish button will not exist after the release. Intead, can we add this line - 'Make your changes as required, then click *Save*, to save each secret type.'

Author: Revised as:

. Make changes to the secrets as required.

As you create or save changes to a secret, the changes are applied immediately. The updated secret group is immediately available the next time the secret group is accessed or when deploying an application.

> The secret group is updated with the changes, and the secrets manager provides the updated secrets to all redeployed applications after this point. 

Reviewer: This sentence is not correct anymore. The save secrets for each type needs an update like that seen below: 

"When each secret is saved, the updated secret is available for use by the next application deployment or the next access of the secret."

Author: Changed to:

After clicking *Finish*, Secrets Manager saves your changes. The updated secret group is immediately available the next time the secret group is accessed or when deploying an application.
